### ISSUE TITLE
Windows path with \x causing migration issues

### ISSUE THREAD

**Issue by Griatch** [evennia/evennia#1599]
#### Brief summary of issue / Description of requested feature:

A user reported an issue on Windows when installing Evennia and running the `migrate` step. The name of the folder was `xavierpc`. Since Windows uses backslashes for paths, this apparently caused the initial `\x` in `...\xavierpc` to be interpreted as a character. The user also had a Japanese locale, it's unclear if this had an effect. Installing Evennia in a different-named folder worked correctly. 

#### Steps to reproduce the issue / Reasons for adding feature:

1. Run installation, install in a folder starting with `x` ?  

#### Error output / Expected result of feature

```
(pyenv) C:\Users\xavierpc\Documents\GitHub\MUD\darkmud>evennia migrate
Traceback (most recent call last):
  File "c:\users\xavierpc\documents\github\mud\pyenv\Scripts\evennia_launcher.py", line 6, in <module>
    exec(compile(open(__file__).read(), __file__, 'exec'))
  File "C:\Users\xavierpc\Documents\GitHub\MUD\evennia\bin\windows\evennia_launcher.py", line 17, in <module>
    main()
  File "c:\users\xavierpc\documents\github\mud\evennia\evennia\server\evennia_launcher.py", line 1320, in main
    init_game_directory(CURRENT_DIR, check_db=check_db)
  File "c:\users\xavierpc\documents\github\mud\evennia\evennia\server\evennia_launcher.py", line 855, in init_game_direc
tory
    django.setup()
  File "c:\users\xavierpc\documents\github\mud\pyenv\Lib\site-packages\django\__init__.py", line 22, in setup
    configure_logging(settings.LOGGING_CONFIG, settings.LOGGING)
  File "c:\users\xavierpc\documents\github\mud\pyenv\Lib\site-packages\django\conf\__init__.py", line 56, in __getattr__
 
    self._setup(name)
  File "c:\users\xavierpc\documents\github\mud\pyenv\Lib\site-packages\django\conf\__init__.py", line 41, in _setup
    self._wrapped = Settings(settings_module)
  File "c:\users\xavierpc\documents\github\mud\pyenv\Lib\site-packages\django\conf\__init__.py", line 110, in __init__
    mod = importlib.import_module(self.SETTINGS_MODULE)
  File "C:\Users\xavierpc\AppData\Roaming\Python\Python27\Lib\importlib\__init__.py", line 37, in import_module
    __import__(name)
ValueError: invalid \x escape
```

#### Extra information, such as Evennia revision/repo/branch, operating system and ideas for how to solve / implement:

It's unclear if this is a Django or Evennia issue - the error appears in the Django initialization step but it might be that Evennia converts to a binary string or simular before.

**Comment by adityac8** [comment]
The same happens with many characters when using `Windows`. I suppose passing them as `unicode` string might be helpful.

**Comment by Griatch** [comment]
@taladan It might be resolved by Python3 yes depending on where the conversion to bytes goes wrong. That said, Evennia already uses unicode for all internal storage so one needs to examine where the conversion confuse Windows. 

*(Note: py3 is not in the dev branch yet, it's in a separate py3 branch that still sits on Evennia 0.6/0.7).*

**Comment by vincent-lg** [comment]
On it.  The way I see it, there are two clear options: either remove the `\` altogether in path (remember that, on Windows as on other OSes, Python can work with a `/` as path separator just fine) or we could force-escape the `\`.  I'll see on what we can have control and what we can play with.

**Comment by vincent-lg** [comment]
This needs to be tested by somebody else, but for my part I don't have this error.  Please notice that some early fixes were brought (notably the `r` being inserted before the `settings.py` docstring to avoid such problems).  This was an early fix in develop-py3 (at the time).  On develop branch with Python 3.7 and Django 2.1 (the default now), I don't have any error.  More to the point, I don't see anything that could possibly go wrong with this setting (no unescaped-string with `\` in my settings file).  Notice, though, this has obviously been fixed a long time ago on develop-py3 (develop) branch and might have been affected (in a good way) by Django transition to 2.X.

Reproduced steps:

1. Clean install (I entirely removed my Python environment and started from scratch with Python 3.7.3 on Windows).
2. Creating a `xo` folder: `evennia --init xo`
3. Running migrations: `evennia migrate` (still no error)
4. Starting Evennia: `evennia start` (still no error)
5. Checking `settings.py` file: nothing wrong. Checking `secret_settings.py` file: nothing wrong.

So I guess you could close it.  If you wish to be careful you could ask someone else to test this very setup or ask the original user.  Perhaps the Japanese locales didn't help, though I don't know how that's relates to the traceback you posted.

(If such error occurred, it would apply to path starting with `n`, `r`, `t` and other symbols in Windows, which  would indeed be problematic.)

**Comment by Griatch** [comment]
@vincent-lg That's very encourating. I'll mark this as resolved then, so it closes with the release of Evennia 0.9!

**Comment by Griatch** [comment]
Closed with merger of develop branch.
